pkg/operator/workloads/data_job.go

```diff
@@ -51,11 +55,122 @@ func dataJobSpec(
 	if shouldIngest {
 		args = append(args, "--ingest")
 	}
-	spec := spark.Spec(workloadID, ctx, workloadTypeData, sparkCompute, args...)
+	spec := Spec(workloadID, ctx, workloadTypeData, sparkCompute, args...)
 	argo.EnableGC(spec)
 	return spec
 }
 
+func Spec(workloadID string, ctx *context.Context, workloadType string, sparkCompute *userconfig.SparkCompute, args ...string) *sparkop.SparkApplication {
```

Comment:
nit: maybe this should be called `sparkSpec()` since it's in the `workloads` package now, and can be private?